### PR TITLE
Update Launcher splashscreen ui

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
Updates the Launcher UI to (mostly) match the Splash Screen PR:
https://github.com/runelite/runelite/pull/3497/files

FuxeLXATfO

I also added a launcher version check commit which will prompt them with the below message and prevent the client from loading if the check doesn't pass.

This is expecting the `bootstrap.json` file hosted at `static.runelite.net` to have an array of strings on the `acceptedLauncherVersions` key. If that array or the launcher version is null the check doesn't run.

<details>
<summary>Old media</summary>
<p>

eGMi8K2FFJ

</p>
</details>

The Splash Screen code can be found here:
https://github.com/TheStonedTurtle/splashscreen

### CODE REVIEW

**Comment by deathbeam** [comment]
Maybe the launcher UI could be separate artifact, and both client and launcher would use it as dependency? I think that would make most sense

**Comment by deathbeam** [comment]
Uh, I mean splash UI

**Comment by Adam-** [comment]
Additionally it would be useful if an error was encountered to pop up a dialog/maybe have a button to open the logfile?

**Comment by TheStonedTurtle** [comment]
The Splash Screen has been moved to a separate repo (https://github.com/TheStonedTurtle/splashscreen) and has a pop-up error dialog with a button linking to the log file.



The splashscreen repo will need reviewing as well.

**Comment by devLotto** [comment]
What do you think of this more spacious concept? The launcher version is also not as prominent in this concept because it's not that important? (but should still be visible somewhere).

runelite_launcher_concept

or

runelite_launcher_concept_2

**Comment by TheStonedTurtle** [comment]
I've updated the splash screen repo to look similar to the 2nd option Lotto suggested.

X7LlGaa2GI

File: src/main/java/net/runelite/launcher/Launcher.java [hunk 1]
```diff
@@ -172,29 +183,64 @@ public static void main(String[] args)
...
 			return;
 		}
 
+		frame.setMessage("Checking launcher version...", 10);
+		final String[] versions = bootstrap.getAcceptedLauncherVersions();
+		final String installedVersion = PROPERTIES.getVersion();
+		if (versions != null && installedVersion != null)
+		{
+			boolean contains = false;
+			for (final String version : versions)
```
Comment (abextm) [inline src/main/java/net/runelite/launcher/Launcher.java]: same thing, this can just be a strcmp with the first good version

**Comment by Adam-** [comment]
Do we have a place for comments on the splashscreen repo? For one, it should not depend on logback

**Comment by TheStonedTurtle** [comment]
I've created a pull request to an empty branch on the splash screen repo we can use for comments/reviews.

https://github.com/TheStonedTurtle/splashscreen/pull/1

**Comment by Adam-** [comment]
This should probably use `Runtime.Version`

File: src/main/java/net/runelite/launcher/Launcher.java [hunk 2]
```diff
@@ -161,13 +167,19 @@ public static void main(String[] args)
 		try
 		{
 			UIManager.setLookAndFeel(UIManager.getCrossPlatformLookAndFeelClassName());
```
Comment (deathbeam) [inline src/main/java/net/runelite/launcher/Launcher.java]: Maybe move this UIManager call inside swing thread too? Or remove it if its unnecessary

File: src/main/java/net/runelite/launcher/Launcher.java [hunk 3]
```diff
@@ -177,29 +189,54 @@ public static void main(String[] args)
...
+		final String launcherVersion = PROPERTIES.getVersion();
+		final Integer minJavaVersion = bootstrap.getMinimumJavaVersion();
+		final Integer javaVersion = getJavaFeatureVersion();
+		if ((minLauncherVersion != null && launcherVersion != null && minLauncherVersion.compareTo(launcherVersion) > 0)
+			|| (minJavaVersion != null && javaVersion != null && minJavaVersion < javaVersion))
+		{
+			try
+			{
+				SwingUtilities.invokeAndWait(frame::invalidVersion);
+				System.exit(0);
```
Comment (deathbeam) [inline src/main/java/net/runelite/launcher/Launcher.java]: This should be non-zero as well. Also, use positive exit codes everywhere, negative means system errors (like interruption of program or forced termination).

**Pull request closed without merging**